toolchain/lexer/testdata/basic_syntax.carbon

```diff
@@ -3,21 +3,22 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 // AUTOUPDATE
-// CHECK:STDOUT: [
 
-// CHECK:STDOUT: { index:  0, kind:                'Fn', line: {{ *}}[[@LINE+7]], column:  1, indent: 1, spelling: 'fn', has_trailing_space: true },
-// CHECK:STDOUT: { index:  1, kind:        'Identifier', line: {{ *}}[[@LINE+6]], column:  4, indent: 1, spelling: 'run', identifier: 0 },
-// CHECK:STDOUT: { index:  2, kind:         'OpenParen', line: {{ *}}[[@LINE+5]], column:  7, indent: 1, spelling: '(', closing_token: 5 },
-// CHECK:STDOUT: { index:  3, kind: 'StringTypeLiteral', line: {{ *}}[[@LINE+4]], column:  8, indent: 1, spelling: 'String', has_trailing_space: true },
-// CHECK:STDOUT: { index:  4, kind:        'Identifier', line: {{ *}}[[@LINE+3]], column: 15, indent: 1, spelling: 'program', identifier: 1 },
-// CHECK:STDOUT: { index:  5, kind:        'CloseParen', line: {{ *}}[[@LINE+2]], column: 22, indent: 1, spelling: ')', opening_token: 2, has_trailing_space: true },
-// CHECK:STDOUT: { index:  6, kind:    'OpenCurlyBrace', line: {{ *}}[[@LINE+1]], column: 24, indent: 1, spelling: '{', closing_token: 10, has_trailing_space: true },
 fn run(String program) {
-  // CHECK:STDOUT: { index:  7, kind:            'Return', line: {{ *}}[[@LINE+3]], column:  3, indent: 3, spelling: 'return', has_trailing_space: true },
-  // CHECK:STDOUT: { index:  8, kind:        'Identifier', line: {{ *}}[[@LINE+2]], column: 10, indent: 3, spelling: 'True', identifier: 2 },
-  // CHECK:STDOUT: { index:  9, kind:              'Semi', line: {{ *}}[[@LINE+1]], column: 14, indent: 3, spelling: ';', has_trailing_space: true },
+// CHECK:STDOUT: [
```

Comment:
This file is worth looking at to see a case where I think things aren't perfect with this scheme, but are generally fine.

The open `[` here is a bit odd, but I don't tihnk we can avoid that w/o something much fancier like `CHECK-LABEL` style things.

And the previous version had some oddities as well, such as the placement of the `]`. I actually prefer the way `EndOfFile` ends up handled here.

(no action needed here)

Reply:
One interesting thing happening in this file (and a few others): we fail to replace the `line: 24` in the `EndOfFile` CHECK, because in general we can't cope with line replacements that refer to a CHECK line. (We can't track them as they get removed and recreated when generating the new test expectations, so we don't know what line is being referred to in the new output.)

Forcing the STDOUT to the end of the file means we run into this every time the output refers to EOF. Perhaps we should have a directive to say "write all remaining CHECK lines here" so that we can make the EOF get left alone in cases where it matters?

Reply:
I've pushed changes to fix the missing line replacements for references to the end of file, and to put the unattached STDOUT checks immediately after the AUTOUPDATE if there are no attached STDOUT checks at all. With that, I'm happy with how the lexer tests are looking.